modulemd/v2/modulemd-module-index.c

```diff
@@ -750,6 +752,150 @@ modulemd_module_index_add_translation (ModulemdModuleIndex *self,
+      for (i = 0; i < streams->len; i++)
+        {
+          stream = g_ptr_array_index (streams, i);
+
+          if (!modulemd_module_index_add_module_stream (
+                into, stream, &nested_error))
+            {
+              g_propagate_error (error, g_steal_pointer (&nested_error));
+              return FALSE;
+            }
+        }
+
+
+      /* Merge any defaults entry for this module */
+      g_debug ("Prioritizer: merging defaults for %s", module_name);
+      defaults = modulemd_module_get_defaults (module);
+      into_defaults = modulemd_module_get_defaults (into_module);
+      if (override)
+        {
+          /* If we've been told to override (we're at a higher priority level),
+           * then just replace the current defaults with the new one
+           */
+          if (!modulemd_module_index_add_defaults (
+                into, defaults, &nested_error))
+            {
+              g_propagate_error (error, g_steal_pointer (&nested_error));
+              return FALSE;
+            }
+        }
+      else if (defaults && !into_defaults)
+        {
+          /* There are no defaults on the target module yet. Copy these */
+          if (!modulemd_module_index_add_defaults (
+                into, defaults, &nested_error))
+            {
+              g_propagate_error (error, g_steal_pointer (&nested_error));
+              return FALSE;
+            }
+        }
+      else
+        {
+          merged_defaults =
+            modulemd_defaults_merge (defaults, into_defaults, &nested_error);
+          if (!merged_defaults)
+            {
+              g_propagate_error (error, g_steal_pointer (&nested_error));
+              return FALSE;
+            }
+
+          /* Add the new, merged defaults to the index */
+          if (!modulemd_module_index_add_defaults (
+                into, merged_defaults, &nested_error))
+            {
+              g_propagate_error (error, g_steal_pointer (&nested_error));
+              return FALSE;
+            }
+          g_clear_object (&merged_defaults);
+        }
+
+      /* Merge translations for this module */
+      g_debug ("Prioritizer: merging translations for %s", module_name);
+      translated_stream_names =
+        modulemd_module_get_translated_streams (module);
+      for (i = 0; i < translated_stream_names->len; i++)
+        {
+          translated_stream_name =
+            g_ptr_array_index (translated_stream_names, i);
+          translation =
+            modulemd_module_get_translation (module, translated_stream_name);
+          trans_stream = modulemd_translation_get_module_stream (translation);
+          current_translation =
+            modulemd_module_get_translation (into_module, trans_stream);
+
+          if (!current_translation ||
+              modulemd_translation_get_modified (translation) >
+                modulemd_translation_get_modified (current_translation))
```

Comment:
There's no merge logic for translations. Unlike Defaults, their `modified` field is authoritative. It would be a change in expected behavior from 1.x to support overrides allowing them to go "backwards". If you think that should be reconsidered, please open an issue on it.